crates/host/src/wasmbus/mod.rs

```diff
@@ -1394,6 +1399,26 @@ impl Deref for Actor {
     }
 }
 
+// NOTE: this is specifically a function instead of an [Actor] method
+// to avoid deadlocks on the instances field.
+/// Returns the first instance that matches the provided annotations
+fn matching_instance(
+    instances: &HashMap<BTreeMap<String, String>, Arc<ActorInstance>>,
+    annotations: &Annotations,
+) -> Option<Arc<ActorInstance>> {
+    // for loop here so we can short circuit
```

Comment:
Nit: `find` should short-circuit too, right?

Reply:
fine I'll make it functional